compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -188,6 +195,44 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
             s"Exported $kind $jsName conflicts with ${alts.head.fullName}")
       }
 
+      genMemberExportOrDispatcher(classSym, jsName, isProp, alts,
+          isDispatcher = false)
+    }
+
+    private def genJSClassDispatcher(classSym: Symbol, name: String): js.Tree = {
+      var alts: List[Symbol] = Nil
+      for {
+        sym <- classSym.info.members
+        if !sym.isBridge && jsNameOf(sym) == name
+      } {
+        val tpe = sym.tpe
+        if (!alts.exists(alt => tpe.matches(alt.tpe)))
+          alts ::= sym
+      }
+
+      assert(!alts.isEmpty,
+          s"Ended up with no alternatives for ${classSym.fullName}::$name.")
+
+      val (propSyms, methodSyms) = alts.partition(jsInterop.isJSProperty(_))
+      val isProp = propSyms.nonEmpty
+
+      if (isProp && methodSyms.nonEmpty) {
+        reporter.error(alts.head.pos,
+            s"Conflicting properties and methods for ${classSym.fullName}::$name.")
+        js.EmptyTree
+      } else if (isProp && alts.exists(_.owner != classSym)) {
+        reporter.error(alts.head.pos,
+            "Illegal override/overload of inherited getter/setter for " +
+            s"${classSym.fullName}::$name. All alternatives must be covered.")
```

Comment:
Ah, this is in fact an implementation restriction. It comes from the inability to call a super getter or super setter. But in fact ES 6 can do it, and [so can Babel](https://github.com/google/traceur-compiler/blob/67180b377559adeb330260e95b878afb99e826af/src/runtime/classes.js#L29-L64). So I'll fix that.